csv-core/src/reader.rs

```diff
@@ -1023,7 +1025,9 @@ impl Reader {
                 }
             }
             InQuotedField => {
-                if self.quoting && self.quote == c {
+                if self.delimiter == c {
+                    (InField, NfaInputAction::CopyToOutput)
```

Comment:
I don't see how this could possibly be right..? This is saying that if the parser is in a quoted field and it sees a delimiter, then it should pop out of the quoted field immediately. But the whole point of quoted fields is to treat the delimiter as just another character. So I don't think this state should be doing any case analysis on the delimiter at all.